Project-Descriptions-and-Plans/LLVM-Toolchain/COREV_LLVM_Plan_Approved_2023.md

```diff
@@ -0,0 +1,173 @@
+
+## Explanation of why OpenHW should do this project
+
+This document should be read in the context of the overall CORE-V LLVM project project launch proposal.  It does not duplicate information from that document.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+This document should be read in the context of the overall CORE-V LLVM project project launch proposal.  It does not duplicate information from that document.
+
+### Market differentiators
+## External dependencies
++ Prequisites:
+    + a suitable platform for regression testing the compiler.
+       * expected to be either OVPSim or the Verilator model of CV32E40Pv2 due in early 2023.
+    + agreement on the instruction set encodings to be compliant with the RISC-V standard.
+       * draft is currently in [this spreadsheet](https://github.com/openhwgroup/cv32e40p/files/9949531/pulp_encoding_blocks-OPHW-2022-10-07.xlsx).
++ External dependencies:
+    + ongoing tracking of upstream LLVM development until the CORE-V tool chain is accepted upstream.
+    + OVPSim licenses from Imperas if this is chosen for regression testing.
+
+## List of project outputs
++ Final deliverables
+    + extensions to upstream LLVM compiler tools to support CORE-V in the upstream llvm-project repository; and
+    + revisions to the CORE-V design specifications to clarify ambiguities.
+
++ Interim deliverables
+    + Reports on progress to the monthly SW TG:
+    + progress against work packages;
+    + regression test results;
+    + updates to the project plan; and
+    + extensions to upstream LLVM compiler tools to support CORE-V in the OpenHW Group llvm-project repository development branch.
+
++ Continuously updated source code as new features are added.
+
+
+### Documentation:
+This will follow the upstream project structure.
+
+### Design
+This will follow the upstream project structure.
+ 
+### Verification:
+This will follow the upsream project structure.  In addition the project will use a subset of the GCC regression tests to verify execution behavior.
+
+### OpenHW engineering staff resource plan: requirement and availability
+
+### Engineering resource supplied by members - requirement and availability
+
+1 Senior engineer from PLCT 
+
+### Marketing resource - requirement and availability
+### Funding for project aspects - requirement and availability
+This project is funded by PLCT
+
+## Architecture and/or context diagrams
+This can be found in the upstream project
+
+## Who would make use of OpenHW output
+
+Everyone who wants to develop software for CORE-V family of cores and devices implementing the CORE-V extensions will need a compiler.  They have a choice to use the CORE-V GCC compiler tool chain or the CORE-V Clang/LLVM tool chain (this project).
+
+
+## Project license model
+
+See License Scheme above.
+
+## Description of initial code contribution, if required
+
+embecosm - V1
```

Comment:
Can you elaborate what is the input to the project from Embecosm? I think the code is a fork of upstream.. does some work come from Embecosm as an inital code contribution?

Reply:
Embecosm will provide advice, patch review and other support through the overall project lead, Charlie Keaney and the Software TG chair, Jeremy Bennett.  The work will also be discussed through the weekly CORE-V LLVM videoconference.

Reply:
The embecosm contribution is already [in the repo](https://github.com/openhwgroup/corev-llvm-project/commits/development), from [e96fda8](https://github.com/openhwgroup/corev-llvm-project/commit/e96fda82f1e93050f413ed1c13640c93dfb80921) to [6c12852](https://github.com/openhwgroup/corev-llvm-project/commit/6c128521db382f74eaf78b5268e6256606c564ff). This contains support for some of the directives, but they may not apply to CV32E40PV2 and will need to be modified.